metrics/prometheus.go

```diff
@@ -537,6 +543,26 @@ const (
 	containerEnvPrefix   = "container_env_"
 )
 
+// DefaultContainerLabels implements ContainerLabelsFunc. It exports the
+// container name, first alias, image name as well as all its env and label
+// values.
+func DefaultContainerLabels(container *info.ContainerInfo) map[string]string {
+	set := map[string]string{"id": container.Name}
```

Comment:
Sure! Any suggestions/pointers where to document that?

On Sun, Aug 21, 2016 at 2:33 AM Jimmi Dyson <notifications@github.com>
wrote:

> In metrics/prometheus.go
> <https://github.com/google/cadvisor/pull/1429#discussion_r75589584>:
>
> > @@ -537,6 +543,26 @@ const (
> >  	containerEnvPrefix   = "container_env_"
> >  )
> >
> > +// DefaultContainerLabels implements ContainerLabelsFunc. It exports the
> > +// container name, first alias, image name as well as all its env and label
> > +// values.
> > +func DefaultContainerLabels(container *info.ContainerInfo) map[string]string {
> > +	set := map[string]string{"id": container.Name}
>
> So this now handles all labels, including id name etc. I personally like
> that control for custom impls but it needs to be documented somewhere as
> well as release noted.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/cadvisor/pull/1429/files/5ad6631b525a0335393344de944427982b26b00e#r75589584>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAANaOkHOd4teDQxJEkc-yPAQXGeSE9zks5qh_FTgaJpZM4JpKV_>
> .
>